ext/curl/interface.c

```diff
@@ -1791,11 +1791,30 @@ static void curl_free_post(void **post)
 }
 /* }}} */
 
-/* {{{ curl_free_stream
+struct mime_data_cb_arg {
+	php_curl *ch;
+	union {
+		zval postfields;
+		struct {
+			zend_string *filename;
+			php_stream *stream;
+		};
+	};
+};
+
+/* {{{ curl_free_cb_arg
  */
-static void curl_free_stream(void **post)
+static void curl_free_cb_arg(void **cb_arg_p)
 {
-	php_stream_close((php_stream *)*post);
+	struct mime_data_cb_arg *cb_arg = (struct mime_data_cb_arg *) *cb_arg_p;
+
+	if (cb_arg->ch) {
+		Z_DELREF(cb_arg->postfields);
```

Comment:
Who owns this? Why is DELREF sufficient?

Reply:
It is owned by this list, so yeah, Z_DELREF is not sufficient. Fixed.